### PR TITLE
 [Intl] Revise timezone name generation

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes (inlcluding intl-data group)
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

This is the final polishing needed for #31294 :)

I've realized it's much easier to de-duplicate by processing fallback locales separate, and then only keep the diff compared to a specific locale. More or less the same approach `LocaleDataGenerator` already follows. I was trying to be clever and filter based on inheritance in a single process; bad idea.

Includes https://github.com/ro0NL/symfony/commit/31591d0 (ref #31432)

### CODE REVIEW

File: src/Symfony/Component/Intl/Resources/data/timezones/en.json [hunk 1]
```diff
@@ -151,7 +151,7 @@
         "America\/Moncton": "Atlantic Time (Moncton)",
         "America\/Monterrey": "Central Time (Monterrey)",
         "America\/Montevideo": "Uruguay Time (Montevideo)",
-        "America\/Montreal": "Canada (Montreal)",
+        "America\/Montreal": "Canada Time (Montreal)",
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Resources/data/timezones/en.json]: use the `regionFormat` for country-based names, this is much more consistent with explicitly translated zones, e.g. `Coordinated Universal Time` and `Central European Time`

File: src/Symfony/Component/Intl/Resources/data/timezones/fr_CA.json [hunk 2]
```diff
@@ -99,7 +99,6 @@
         "Indian\/Mayotte": "heure d’Afrique orientale (Mayotte)",
         "Indian\/Reunion": "heure de la Réunion",
         "PST8PDT": "heure du Pacifique",
-        "Pacific\/Easter": "heure de l’île de Pâques",
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Resources/data/timezones/fr_CA.json]: here's the bugfix; this should be inherited from `fr` locale

File: src/Symfony/Component/Intl/Resources/data/timezones/root.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "Version": "2.1.49.82",
+    "Meta": {
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Resources/data/timezones/root.json]: root metadata is now compiled into root.json, in order to de-duplicte locale metadata

File: src/Symfony/Component/Intl/Resources/data/timezones/rw.json [hunk 4]
```diff
@@ -3,20 +3,8 @@
     "Names": {
         "Africa\/Kigali": "U Rwanda (Kigali)",
         "Antarctica\/Troll": "Troll",
-        "CST6CDT": "CST6CDT",
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Resources/data/timezones/rw.json]: we should never provide zone IDs as translations, in this case the runtime fallback should apply instead.

File: src/Symfony/Component/Intl/Resources/data/timezones/zh_HK.json [hunk 5]
```diff
@@ -1,202 +0,0 @@
-{
-    "Version": "2.1.47.86",
-    "Names": {
-        "Africa\/Abidjan": "格林尼治标准时间（阿比贊）",
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Resources/data/timezones/zh_HK.json]: only alias locales are deleted (related to #31411)

File: src/Symfony/Component/Intl/Resources/data/timezones/bs.json [hunk 6]
```diff
@@ -438,9 +438,6 @@
     },
     "Meta": {
         "GmtFormat": "GMT %s",
-        "HourFormat": [
-            "+%02d:%02d",
-            " -%02d:%02d"
-        ]
+        "HourFormatNeg": " -%02d:%02d"
```
Comment (fabpot) [inline src/Symfony/Component/Intl/Resources/data/timezones/bs.json]: missing HourFormatPos?

Comment (ro0NL) [reply src/Symfony/Component/Intl/Resources/data/timezones/bs.json]: it'll be inherited from root.json (`"HourFormatPos": "+%02d:%02d"`), which is the same format as `bs` locale (`+HH:mm`)

i've no idea why the negative format has a leading space :man_shrugging: (upstream data), these cases shoud be confirmed by native speakers IMHO.

this basically shows the de-duplication works :D

**Comment by fabpot** [comment]
Can you rebase to get green tests?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Ready for rebase on 4.3.

**Comment by ro0NL** [comment]
Done. (failures unrelated)

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Pull request merged**